gcc/rust/typecheck/rust-hir-type-check.h

```diff
@@ -79,6 +82,49 @@ class TypeCheckContextItem
   Item item;
 };
 
+/**
+ * Interned lifetime representation in TyTy
+ *
+ * On the HIR->TyTy boundary HIR::Lifetime is interned into this struct.
+ */
+class Lifetime
```

Comment:
I dont think you need to change this now but we already have AST::Lifetime and HIR::Lifetime, I wonder should this really be moved to its own namespace at some point Resolver is ok but not sure its the right name.

Reply:
My thinking for the future was that the interning might be pushed up to AST to avoid passing the strings around. Then this class would be pointless.
Right now it has only very short life - it is used to resolve lifetimes (syntactic objects) into regions (semantic objects). All of it use happens in the resolver.